src/main/java/spring/intro/dao/UserDaoImpl.java

```diff
@@ -38,6 +39,19 @@ public void add(User user) {
         }
     }
 
+    @Override
+    public Optional<User> findById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            Query<User> query = session.createQuery("from User "
+                    + "where id = :id", User.class);
+            query.setParameter("id", id);
+            return query.uniqueResultOptional();
```

Comment:
It's only a suggestion to use CriteriaAPI.
I suppose it works faster for select queries. 🤔

Reply:
yes, it's optional